test-utils/src/dev/falcon.rs

```diff
@@ -4,10 +4,62 @@
 
 //! Example test to run in a VM.
 
-#[cfg(all(test, target_os = "illumos"))]
 mod test {
+    use super::*;
+    use crate::dev::test_setup_log;
     use anyhow::anyhow;
     use gethostname::gethostname;
+    use libfalcon::unit::gb;
+
+    #[ignore]
+    #[tokio::test]
+    async fn run_launch_in_vm() {
+        let args: Vec<_> = std::env::args().collect();
+        let logctx = test_setup_log("launch");
+        let log = logctx.log.clone();
+        let runner_name = "launchpad_mcduck";
+        let node_name = "launchpad_mcduck_test_vm";
+        let mut test = super::FalconTest::new(runner_name, &log).unwrap();
+        eprintln!("args = {:?}", args);
+        // Create a falcon node (VM)
+        let vm = test.runner().node(node_name, "helios-2.0", 1, gb(2));
+
+        // Create a place(cargo-bay) to store data to be mounted into the VM
+        let cargo_bay = tempdir().unwrap();
+        info!(log, "Set cargo-bay to {} on host machine", cargo_bay.path());
+        let test_name = "launch";
+        let source_test_path = &args[0];
+        let test_file_name = camino::Utf8Path::new(source_test_path)
+            .file_name()
+            .expect("Failed to get test file name");
+        let test_path = cargo_bay.path().to_path_buf().join(test_file_name);
+
+        // Copy the test binary into the cargo-bay
+        info!(log, "Copying {source_test_path} to {test_path}");
+        std::fs::copy(camino::Utf8Path::new(source_test_path), &test_path)
+            .expect("failed to copy source test binary into cargo-bay");
+
+        // Mount the cargo-bay into the VM
+        test.runner()
+            .mount(cargo_bay.path(), "/opt/cargo-bay", vm)
```

Comment:
"/opt/cargo-bay", and the host path are both arbitrary, and can be anything. As an example, if a test wanted to launch multiple VMs they may all use the same VM mount path, but have different local directories. This is part of the reason for allowing full access to the falcon API.